cinema/serializers.py

```diff
@@ -68,13 +79,87 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available",
         )
 
 
+class TicketSerializer(serializers.ModelSerializer):
+    movie_session = serializers.PrimaryKeyRelatedField(queryset=MovieSession.objects.all())
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "movie_session", "row", "seat")
+
+    def validate(self, attrs):
+        movie_session = attrs.get("movie_session")
+        row = attrs.get("row")
+        seat = attrs.get("seat")
+
+        if not movie_session:
+            raise serializers.ValidationError(
+                {"movie_session": "Movie session is required to validate row and seat."}
+            )
+
+        cinema_hall = movie_session.cinema_hall
```

Comment:
The `validate` method in `TicketSerializer` assumes that `movie_session` is a fully populated instance with a `cinema_hall` attribute. Ensure that `movie_session` is fully populated or adjust the logic to handle cases where it might not be.